packages/react-i18n/src/i18n.ts

```diff
@@ -343,18 +343,21 @@ export class I18n {
     return WEEK_START_DAYS.get(country) || DEFAULT_WEEK_START_DAY;
   }
 
-  getCurrencySymbol = (currencyCode?: string) => {
+  getCurrencySymbol = (currencyCode?: string, locale: string = this.locale) => {
```

Comment:
@trishrempel: A null check will still be needed somewhere: either in `getCurrencySymbol` or `getShortCurrencySymbol`. Otherwise, users will be able to call `formatCurrency` without entering a currency value. This is because `defaultCurrency` in the constructor is optional ([see here](https://github.com/Shopify/quilt/blob/f6ba8931b7ce598b3102e1c720c9ce47e7237776/packages/react-i18n/src/i18n.ts#L75)).